### PR TITLE
Dropdown changes to storage capacity-card

### PR DESCRIPTION
**Pull request by aruniiird**
Dropdown functionality added to the storage-dashboard capacity card

Signed-off-by: aruniiird <arun.iiird@gmail.com>

### CODE REVIEW

**Comment by aruniiird** [comment]
@cloudbehl , @rawagner , all the needed changes have gone into, please take a look...
I am unable to run the queries, as I could see (in the chrome-debug mode), are failing to connect to the prometheuse server... Any pointers will be helpful, then I can put working card images here...

**Comment by aruniiird** [comment]
@cloudbehl , All the changes suggested has gone in.

a. Usage of keys() (instead of a map() function)
b. Variable names (like qResultTotal / qResultUsed changed)
c. Removed the complex map, and used simple variables instead to hold the prometheus result objects
d. Removed the null checks on prometheus result objects

Screen shots,

VMs_vs_Pods
Requested_vs_Used
total_capacity_card

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 1]
```diff
@@ -24,42 +24,79 @@ const getLastStats = (response, getStats: GetStats): React.ReactText => {
...
-  }, [watchPrometheus, stopWatchPrometheusQuery]);
+      matchingQueries.forEach((mq) => {
+        stopWatchPrometheusQuery(mq);
+      });
+  }, [watchPrometheus, stopWatchPrometheusQuery, cvTypeSelected]);
+
+  const matchingQueries = QueriesMatchingCapacityView[CapacityViewType[cvTypeSelected]];
+  // 'matchingQueries[0] => TOTAL query
+  // 'matchingQueries[1] => USED query
+  const queryResultStorageTotal = prometheusResults.getIn([matchingQueries[0], 'result']);
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]:
```suggestion
  const storageTotal = prometheusResults.getIn([matchingQueries[0], 'result']);
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 2]
```diff
@@ -24,42 +24,79 @@ const getLastStats = (response, getStats: GetStats): React.ReactText => {
...
+  const queryResultStorageUsed = prometheusResults.getIn([matchingQueries[1], 'result']);
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]:
```suggestion
  const storageUsed = prometheusResults.getIn([matchingQueries[1], 'result']);
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 3]
```diff
@@ -24,42 +24,79 @@ const getLastStats = (response, getStats: GetStats): React.ReactText => {
...
+          selectedKey={[cvTypeSelected]}
+        />
       </DashboardCardHeader>
       <DashboardCardBody>
         <CapacityBody>
           <CapacityItem
             title="Storage"
-            used={getLastStats(storageUsed, getRangeVectorStats)}
-            total={getLastStats(storageTotal, getInstantVectorStats)}
+            used={getLastStats(queryResultStorageUsed, getRangeVectorStats)}
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]:
```suggestion
            used={getLastStats(storageUsed, getRangeVectorStats)}
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 4]
```diff
@@ -24,42 +24,79 @@ const getLastStats = (response, getStats: GetStats): React.ReactText => {
...
+            total={getLastStats(queryResultStorageTotal, getInstantVectorStats)}
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]:
```suggestion
            total={getLastStats(storageTotal, getInstantVectorStats)}
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx [hunk 5]
```diff
@@ -24,42 +24,79 @@ const getLastStats = (response, getStats: GetStats): React.ReactText => {
...
             formatValue={humanizeBinaryBytesWithoutB}
-            isLoading={!(storageUsed && storageTotal)}
+            isLoading={!(queryResultStorageUsed && queryResultStorageTotal)}
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]:
```suggestion
            isLoading={!(storageUsed && storageTotal)}
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx (hunk 1 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: Done...

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx (hunk 2 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-card/capacity-card.tsx]: Done...

**Pull request merged**